audio-format-service/src/main.py

```
@@ -0,0 +1,361 @@
+        )
+    return audio_handler
+
+
+@app.get("/health", response_model=HealthResponse)
+async def health_check():
+    """Health check endpoint."""
+    return HealthResponse(
+        status="healthy",
+        version="1.0.0",
+        dependencies={
+            "audio_handler": "initialized" if audio_handler else "not_initialized",
+            "ffmpeg": "available",  # TODO: Check actual ffmpeg availability
+        }
+    )
+
+
+@app.get("/formats", response_model=List[str])
+async def get_supported_formats(
+    handler: AudioFormatHandler = Depends(get_audio_handler)
+):
+    """Get list of supported audio formats."""
+    return handler.get_supported_formats()
+
+
+@app.post("/validate", response_model=FormatValidationResponse)
+async def validate_format(
+    file: UploadFile = File(...),
+    expected_format: Optional[str] = None,
+    handler: AudioFormatHandler = Depends(get_audio_handler)
+):
+    """Validate audio file format."""
+    if not file.filename:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="File must have a filename"
+        )
+
+    # Save uploaded file temporarily
+    with tempfile.NamedTemporaryFile(delete=False, suffix=f"_{file.filename}") as temp_file:
+        content = await file.read()
+        temp_file.write(content)
+        temp_file_path = temp_file.name
+
+    try:
+        # Validate format
+        is_valid = handler.validate_format(temp_file_path, expected_format)
+        detected_format = handler.detect_format(temp_file_path) if is_valid else None
+
+        return FormatValidationResponse(
+            valid=is_valid,
+            format=detected_format,
+            message="Format validation successful" if is_valid else "Format validation failed"
+        )
+
+    except (AudioFormatError, ValidationError) as e:
+        return FormatValidationResponse(
+            valid=False,
+            format=None,
+            message=str(e)
```

Kiro comment: 🤔 Good observation about inconsistency. However, `/validate` endpoint is designed to return validation results (success/failure) in response body, while `/detect` and `/metadata` should fail fast with HTTP errors.

This is intentional - validation endpoint tells you "is this valid?" while others assume valid input. Could document this in API docs. Thoughts?